### PR TITLE
[nad-viewer] Update text nodes moving and creation to powsybl diagram v. 5.3.0

### PR DESCRIPTION
**Pull request by massimo-ferraro**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been updated (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] A PR or issue has been opened in all impacted repositories (if any)

**Does this PR already have an issue describing the problem?**
no

**What kind of change does this PR introduce?**
feature

**What is the current behavior?**
The network area diagram viewer supports powsybl diagram version 5.2, where all text nodes are included in a single foreign object

**What is the new behavior (if this is a feature change)?**
The network area diagram viewer supports moving and creation of text nodes in diagrams where there is a foreign object for each text node, as in powsybl diagram version 5.3

**Does this PR introduce a breaking change or deprecate an API?**
- [X] Yes
- [ ] No

**What changes might users need to make in their application due to this PR?**
The new code supports diagram version 5.3, SVGs created with old versions of diagram could be not supported (if they have a single foreign object for the text nodes)

### CODE REVIEW

**Review by rolnico — CHANGES_REQUESTED** [review]
There is a change missing, related to the renaming of dangling lines to boundary lines.

I'm wondering if and how we could keep the retro-compatibility with SVG generated on previous versions of diagram... 
@flo-dup what do you think of this? Do we usually keep it or do we accept breaking changes?

**Comment by massimo-ferraro** [comment]
> There is a change missing, related to the renaming of dangling lines to boundary lines.
> 
> I'm wondering if and how we could keep the retro-compatibility with SVG generated on previous versions of diagram... @flo-dup what do you think of this? Do we usually keep it or do we accept breaking changes?

About the renaming of dangling lines to boundary lines, not sure if this have impacts in this code. There is a class name that was changed (nad-dangling-line-edge vs. nad-boundary-line-edge), but this class is not defined in style.css file, nor it is added to the SVG or referred by the network viewer code (this change will impact the SVG creation from metadata, that is now in the integration branch). Where do you see the need to take into account this renaming?

About the retro-compatibility, now we don't have the diagram version in SVG or metadata. If it is needed, in order to handle the different versions of the diagram some heuristic should be used (like checking the dom element of the text node section?). Or there is some other way?

**Comment by massimo-ferraro** [comment]
> > There is a change missing, related to the renaming of dangling lines to boundary lines.
> 
> About the renaming of dangling lines to boundary lines, not sure if this have impacts in this code. There is a class name that was changed (nad-dangling-line-edge vs. nad-boundary-line-edge), but this class is not defined in style.css file, nor it is added to the SVG or referred by the network viewer code (this change will impact the SVG creation from metadata, that is now in the integration branch). Where do you see the need to take into account this renaming?

Sorry, you are right, there is a reference to dangling lines, even if I'm not sure it is used. I renamed it, using boundary lines

**Comment by rolnico** [comment]
> About the renaming of dangling lines to boundary lines, not sure if this have impacts in this code. There is a class name that was changed (nad-dangling-line-edge vs. nad-boundary-line-edge), but this class is not defined in style.css file, nor it is added to the SVG or referred by the network viewer code (this change will impact the SVG creation from metadata, that is now in the integration branch). Where do you see the need to take into account this renaming?
> 
> About the retro-compatibility, now we don't have the diagram version in SVG or metadata. If it is needed, in order to handle the different versions of the diagram some heuristic should be used (like checking the dom element of the text node section?). Or there is some other way?

There will be some impact: in the metadata at least, you will get `BoundaryLineEdge` instead of `DanglingLineEdge`.
(See for example the change in demo/src/diagram-viewers/data/nad-scada_metadata.json in this PR)

Also, we have to be consistent in our naming conventions in the whole powsybl project. Therefore, all the references to `DanglingLine` have to be changed to `BoundaryLine`, for example in `EdgeType`, `EQUIPMENT_TYPES`, `FEEDER_COMPONENT_TYPES`, etc.

**Comment by massimo-ferraro** [comment]
> > About the renaming of dangling lines to boundary lines, not sure if this have impacts in this code. There is a class name that was changed (nad-dangling-line-edge vs. nad-boundary-line-edge), but this class is not defined in style.css file, nor it is added to the SVG or referred by the network viewer code (this change will impact the SVG creation from metadata, that is now in the integration branch). Where do you see the need to take into account this renaming?
> 
> There will be some impact: in the metadata at least, you will get `BoundaryLineEdge` instead of `DanglingLineEdge`. (See for example the change in demo/src/diagram-viewers/data/nad-scada_metadata.json in this PR)
> 
> Also, we have to be consistent in our naming conventions in the whole powsybl project. Therefore, all the references to `DanglingLine` have to be changed to `BoundaryLine`, for example in `EdgeType`, `EQUIPMENT_TYPES`, `FEEDER_COMPONENT_TYPES`, etc.

Renamed dangling line to boundary line also in the SLD viewer

**Review by rolnico — CHANGES_REQUESTED** [review]
Change missing in the file `network-map-layers/src/equipment-types.ts`:

line 31, from `DANGLING_LINE = 'DANGLING_LINE'` to `BOUNDARY_LINE = 'BOUNDARY_LINE'`

**Comment by massimo-ferraro** [comment]
> Change missing in the file `network-map-layers/src/equipment-types.ts`:
> 
> line 31, from `DANGLING_LINE = 'DANGLING_LINE'` to `BOUNDARY_LINE = 'BOUNDARY_LINE'`

Renamed dangling line to boundary line also in map viewer

File: demo/src/diagram-viewers/add-diagrams.ts [hunk 1]
```diff
@@ -264,6 +264,8 @@ export const addNadToDemo = () => {
                 onToggleHoverCallback: handleToggleNadHover,
                 onRightClickCallback: handleRightClick,
                 onBendLineCallback: handleLineBending,
+                enableAdaptiveTextZoom: true,
```
Comment (rolnico) [inline demo/src/diagram-viewers/add-diagrams.ts:269]: There is an issue with this case: the double arrow (new functionnality from diagram) is not managed well

**Review by rolnico — CHANGES_REQUESTED** [review]

File: demo/src/diagram-viewers/add-diagrams.ts (hunk 1 above)
Comment (massimo-ferraro) [reply demo/src/diagram-viewers/add-diagrams.ts:269]: Added handling of double arrows in nad viewer node moving and adaptive zoom functionality

**Comment by rolnico** [comment]
Note: handling the edge info components will be done in the PR #384

**Comment by sBouzols** [comment]
Hi, I tested this PR in GridSuite and it seems ok.
I need more tests for ~dangling~boundary lines cases, will do tomorrow.
@rolnico could you add this css change snippet in the migration guide of powsybl-diagram v5.3.0 ?
https://github.com/powsybl/powsybl-network-viewer/pull/378/files#diff-151e510b12bef62f4e56de2db14a5d9bbfbe45346488bba5e8c081ca79fdb051

**Comment by rolnico** [comment]
> Hi, I tested this PR in GridSuite and it seems ok. I need more tests for ~dangling~boundary lines cases, will do tomorrow. @rolnico could you add this css change snippet in the migration guide of powsybl-diagram v5.3.0 ? https://github.com/powsybl/powsybl-network-viewer/pull/378/files#diff-151e510b12bef62f4e56de2db14a5d9bbfbe45346488bba5e8c081ca79fdb051

Done [here](https://github.com/powsybl/powsybl-diagram/wiki/Migration-guide-v5.3.0#foreign-objects)

**Review by rolnico — APPROVED** [review]

**Comment by sBouzols** [comment]
I did more tests for ~dangling~boundary lines cases with NAD, SLD viewers and the map component.
I didn't catch any regression it's ok then for GridSuite !
Thanks for the migration guide update !

**Review by sBouzols — APPROVED** [review]
